glass/galaxies.py

```diff
@@ -140,7 +140,9 @@ def redshifts_from_nz(
         )
         total += count_out[k]
 
-    assert total == redshifts.size  # noqa: S101
+    if total != redshifts.size:
+        msg = "Total number of sampled redshifts does not match the expected size."
+        raise ValueError(msg)
```

Comment:
This is really an assert because it should never fail.

Reply:
Will revert